### DISCUSSION TITLE
Redirect route using attributes

### DISCUSSION THREAD

**Discussion by eerison** [symfony/symfony#54171] [Q&A] [answered]
Hello guys

Is there any way to redirect old url to new url using route attributes only?

e.g:

```php
    #[Route('/my:old:url', name: 'legacy_foo, methods: ['GET'])]
    #[Route('/my/new/url, name: 'foo', methods: ['GET'])]
```

I saw that exist [Symfony\Bundle\FrameworkBundle\Controller\RedirectController](https://symfony.com/doc/5.x/routing.html#redirecting-to-urls-and-routes-directly-from-a-route), But I didn't see any example using `#[Route]`.

in case it is not possible, then I will check out a listener on request.

thank you.

**Comment by MatTheCat** [comment]
Hello,

AFAIK you cannot use the `Route` attribute to perform a redirect.

> I will check out a listener on request

I would suggest to follow the documentation instead: https://symfony.com/doc/current/routing.html#redirecting-to-urls-and-routes-directly-from-a-route

**Reply by eerison** [reply, marked as answer]
thinking again, I guess create a file with redirects is more interesting. because after I move all urls to the new on.

I can remove  old redirects from one place only, instead of keep searching into the code.

Note: in my case all routes are internal, then SEO isn't an issue for me :)


But just to point out my current approach

````php
#[Route('/dashboard:basic', name: 'dashboard_basic_old', methods: ['GET'], defaults: ['redirectRoute' => 'dashboard_basic'])]
#[Route('/dashboard/basic', name: 'dashboard_basic', methods: ['GET'])]
```

my event

```php
<?php

namespace App\EventListener;

use Symfony\Component\HttpFoundation\RedirectResponse;
use Symfony\Component\HttpKernel\Event\RequestEvent;
use Symfony\Component\Routing\Generator\UrlGeneratorInterface;

class RedirectLegacyUrlListener
{
    public function __construct(private UrlGeneratorInterface $urlGenerator)
    {
    }

    public function __invoke(RequestEvent $event): void
    {
        $request = $event->getRequest();
        $redirectRoute = $request->get('redirectRoute');

        if (null === $redirectRoute) {
            return;
        }

        $url = $this->urlGenerator->generate($redirectRoute);

        $event->setResponse(new RedirectResponse($url));
        $event->stopPropagation();
    }
}
```

**Reply by MatTheCat** [reply]
Well it’s generally not recommended to hack a solution when there is a documented one.

Also I don’t understand why you’re mentioning a single file? This is clearly not the case in the example you provided, and will only get worse the more routes you have. Which is a shame because you can indeed group all your redirects in a single file using the documented solution.

**Reply by eerison** [reply]
> Which is a shame because you can indeed group all your redirects in a single file using the documented solution.

I will group all redirects into a single file as suggested here: https://symfony.com/doc/current/routing.html#rendering-a-template-directly-from-a-route

the example that I provided was just to expose how I was doing ....

keep redirect in one file will be better, because it will be removed later ..., then it is better to have this grouped in one file as the doc suggest!

**Reply by MatTheCat** [reply]
Okaay didn’t understand you were going to follow the documentation since you accepted your own reply including your listener..!
